File: exp1/adv_1_1.py
def get_material(ttl: int, fmt1: int, fmt2: int) -> list:
    res = []

    num1_max = ttl // fmt1
    for index in range(num1_max + 1):
        num2 = (ttl - (index * fmt1)) // fmt2
        remain = ttl - (fmt1 * index + fmt2 * num2)
        if remain < fmt1 and remain < fmt2:

            res.append([index, num2, remain])

    return res
    

File: exp1_ProgrammingBasics/adv_1_1.m
function res = adv_1_1(ttl, fmt1, fmt2)

    res = [];
    num1_max = floor(ttl / fmt1);
    for index = 0:num1_max
        num2 = floor((ttl - (index * fmt1)) / fmt2);
        remain = ttl - (fmt1 * index + fmt2 * num2);
        if remain < fmt1 && remain < fmt2
            res = [res; index, num2, remain];
        end
    end
end